src/app/data-model-provider/MetadataTypes.h

```diff
@@ -87,15 +98,98 @@ enum class AttributeQualityFlags : uint32_t
 struct AttributeEntry
 {
     AttributeId attributeId;
-    BitFlags<AttributeQualityFlags> flags;
 
-    // read/write access will be missing if read/write is NOT allowed
-    //
-    // NOTE: this should be compacted for size
-    std::optional<Access::Privilege> readPrivilege;  // generally defaults to View if readable
-    std::optional<Access::Privilege> writePrivilege; // generally defaults to Operate if writable
+    // Constructor
+
+    StartBitFieldInit // Disabling '-Wconversion' & '-Wconversion'
+
+        constexpr AttributeEntry(
+            AttributeId id = 0,                                                                 // attributeId initial value,
+                                                                                                // this could be altered later
+            AttributeQualityFlags attrQualityFlags     = static_cast<AttributeQualityFlags>(0), // mask.flags initial value
+            std::optional<Access::Privilege> readPriv  = static_cast<Access::Privilege>(0),     // mask.readPrivilege initial value
+            std::optional<Access::Privilege> writePriv = static_cast<Access::Privilege>(0)      // mask.writePrivilege initial value
+            ) :
+        attributeId{ id },
+        mask{ to_underlying(attrQualityFlags) & ((1 << 7) - 1), // Narrowing expression to 7 bits
+              to_underlying(readPriv.value_or(static_cast<Access::Privilege>(0))) &
```

Comment:
the value or static_casts an invalid value. How about?

```suggestion
              (readPriv.has_value() ? to_underlying(*readPriv) : 0) & ((1 << 5) - 1);
```

I think we should add some constexpr constants for these 5 and 7 bits, like kPrivilegeBits and kQualityBits.

Reply:
Yes, `constexpr` values are better and more maintainability.

Something like:

```
// constants used to narrow binary expressions
constexpr uint8_t kPrivilegeBits   = 5;
constexpr uint8_t kAttrQualityBits = 7;
constexpr uint8_t kCmmdQualityBits = 3;
```
Done.